vtex-apps/add-to-cart-button
# Fix/assembly opts

#### What problem is this solving?
Fix the `assemblyOption` structure that is passed to addToCart query. 

#### How to test it?
[Workspace](https://brasileirovtex--acctglobal.myvtex.com/mcproductrefid136/p)

1. Go into workspace
2. Click to customize combo 1
3. Select `Lanche Genérico`
4. Repeat steps 2 and 3 to the other assembly option
5. Click to add product to cart
6. Look at minicart if `Lanche Generico` appears

#### Screenshots or example usage:
**After**:
![image](https://user-images.githubusercontent.com/15680320/106001543-d92a9080-608e-11eb-8753-3f929d4090da.png)
assemblyOptions that's passed to addToCart query
```json
"items": [
        {
            "id": 10104,
            "index": 0,
            "quantity": 1,
            "seller": "1",
            "options": [
                {
                    "assemblyId": "Family Box - 1_Combo2",
                    "id": "10101",
                    "quantity": 1,
                    "seller": "1",
                    "options": [
                        {
                            "assemblyId": "Combo 1_Lanche",
                            "id": "10091",
                            "quantity": 0,
                            "seller": "1"
                        },
                        {
                            "assemblyId": "Combo 1_Lanche",
                            "id": "10093",
                            "quantity": 1,
                            "seller": "1"
                        }
                    ]
                },
             ....
      }
    ]
```
**Before**:
![image](https://user-images.githubusercontent.com/15680320/106001543-d92a9080-608e-11eb-8753-3f929d4090da.png)
```json
"items": [
        {
            "id": 10104,
            "index": 0,
            "quantity": 1,
            "seller": "1",
            "options": [
                {
                    "assemblyId": "Family Box - 1_Combo2",
                    "id": "10101",
                    "quantity": 1,
                    "seller": "1"
                },
                ...,
                {
                    "assemblyId": "Combo 1_Lanche",
                    "id": "10091",
                    "quantity": 0,
                    "seller": "1"
                },
                ...
            ]
        }
    ],
```

#### Describe alternatives you've considered, if any.
Refactor the structure of Assembly Options Groups and Items at the Product Customizer

#### How does this PR make you feel? [:link:](http://giphy.com/)
![](https://media.giphy.com/media/L0Z4qwdwv62cn4haFp/giphy.gif)
